graspologic/embed/case.py

```diff
@@ -0,0 +1,209 @@
+        results if the graph is unconnected. Not checking for connectedness may
+        result in faster computation.
+
+
+    References
+    ---------
+    .. [1] Binkiewicz, N., Vogelstein, J. T., & Rohe, K. (2017). Covariate-assisted
+    spectral clustering. Biometrika, 104(2), 361-377.
+    """
+
+    def __init__(
+        self,
+        alpha: Optional[float] = None,
+        assortative: bool = True,
+        n_components: Optional[int] = None,
+        n_elbows: int = 2,
+        check_lcc: bool = False,
+    ):
+        super().__init__(
+            n_components=n_components,
+            n_elbows=n_elbows,
+            check_lcc=check_lcc,
+            concat=False,
+            algorithm="eigsh",
+        )
+
+        if not isinstance(assortative, bool):
+            msg = "embedding_alg must be a boolean value."
+            raise ValueError(msg)
+        self.assortative = assortative
+
+        if not ((alpha is None) or isinstance(alpha, (float, int))):
+            msg = "alpha must be in {None, float, int}."
+            raise TypeError(msg)
+
+        self.alpha = alpha
+        self.latent_right_ = None
+        self.is_fitted_ = False
+
+    def fit(
+        self, network: Tuple[np.ndarray, np.ndarray], y: None = None
+    ) -> "CovariateAssistedEmbed":
+        """
+        Fit a CASE model to an input graph, along with its covariates. Depending on the
+        embedding algorithm, we embed
+
+        .. math:: L_ = LL + \alpha YY^T
+        .. math:: L_ = L + \alpha YY^T
+
+        where :math:`\alpha` is a tuning parameter which makes the leading eigenvalues
+        of the two summands the same. Here, :math:`L` is the regularized
+        graph Laplacian, and :math:`Y` is a matrix of covariates for each node.
+
+        Covariates are row-normalized to unit l2-norm.
+
+        Parameters
+        ----------
+        network : tuple or list of np.ndarrays
+            Contains the tuple (A, Y), where A is an adjacency matrix and Y is the
+            matrix of covariates.
+
+            A : array-like or networkx.Graph
+                Input graph to embed. See graspologic.utils.import_graph
+
+            Y : array-like, shape (n_vertices, n_covariates)
+                Covariate matrix. Each node of the graph is associated with a set of
+                `d` covariates. Row `i` of the covariates matrix corresponds to node
+                `i`, and the number of columns are the number of covariates.
+
+        y: Ignored
+
+        Returns
+        -------
+        self : object
+            Returns an instance of self.
+        """
+
+        # setup
+        if not isinstance(network, (tuple, list)):
+            msg = "Network should be a tuple-like object of (graph, covariates)."
+            raise TypeError(msg)
+        if len(network) != 2:
+            msg = "Network should be a tuple-like object of (graph, covariates)."
+            raise ValueError(msg)
+
```

Comment:
Do you need to add checks here to make sure the adjacency matrix/covariate matrix are square?

Reply:
the covariate matrix doesn't need to be square (and most of the time won't be)
I used to have a check for square, but pedigo pointed out that it already happens in `import_graph`, so the check was redundant.

Reply:
Ok, that makes sense